torch/distributed/distributed_c10d.py

```diff
@@ -4780,6 +4780,11 @@ def barrier(
         None, if not async_op or if not part of the group
 
     .. note:: `ProcessGroupNCCL` now blocks the cpu thread till the completion of the barrier collective.
+    .. note:: `ProcessGroupNCCL` implements barrier as an all_gather of a 1-element tensor. A device must be chosen
```

Comment:
I think it should be all_reduce (https://github.com/pytorch/pytorch/blob/main/torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp#L5115)